zipkin-lens/src/constants/api.js

```diff
@@ -13,8 +13,30 @@
  */
 const { API_BASE } = process.env;
 
-export const ZIPKIN_API = `${API_BASE || ''}/zipkin/api/v2`;
-export const UI_CONFIG = `${API_BASE || ''}/zipkin/config.json`;
+const inferBasePath = () => {
+  const { pathname } = window.location;
+  // Infer the path that zipkin is mounted on based on the current path.
+  if (pathname.endsWith('/dependency')) {
+    return pathname.substring(0, pathname.length - '/dependency'.length);
+  }
+  if (pathname.endsWith('/traceViewer')) {
+    return pathname.substring(0, pathname.length - '/traceViewer'.length);
+  }
+  const tracesIndex = pathname.lastIndexOf('/traces/');
+  if (tracesIndex !== -1) {
+    return pathname.substring(0, tracesIndex);
+  }
+
+  // Zipkin server always redirects from /zipkin to /zipkin/ but handle the non-redirected path too
+  // just in case.
+  return pathname.endsWith('/') ? pathname.substring(0, pathname.length - 1) : pathname;
```

Comment:
I think it is healthy to just avoid weird routing issues by just removing all `/` at the end using something like https://stackoverflow.com/a/6680877 instead of only looking at the last one. What do you think?

Reply:
Here we're only inferring the base path rather than a full routing - I think that means that less sanitization is better than more since such extraneous `/` removal won't happen within react-router, which essentially concatenates the base path and paths to do routing so I've stuck with the simple match. Added a comment about it too. Let me know if this doesn't make sense.